jccli/cli/group.py

```diff
@@ -14,29 +22,159 @@ def group(ctx):
+        sys.exit(1)
+    try:
+        api2.create_group(name, type)
+    except ApiException:
+        logger.error(f"API error (confirm that no group of type '{type}' and name '{name}' already exists)")
+        sys.exit(1)
+    logger.info(f"successfully created group: {name}")
+
+
+@group.command('get')
+@click.option('--name', '-n', required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
+@click.pass_context
+def get_group(ctx, name, type):
+    """
+    Get detail view of a group
+    """
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
     logger = ctx.obj.get('logger')
-    response = api2.create_group(name, type)
-    logger.info(f"{response}")
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    response = api2.get_group(group_name=name, group_type=type)
+    serialized_response = json.dumps(response)
+    logger.info(f"{serialized_response}")
+
+
+@group.command('list')
+@click.option('--user', 'type', flag_value='user_group', help='Restrict to user groups only')
+@click.option('--system', 'type', flag_value='system_group', help='Restrict to system groups only')
+@click.pass_context
+def list_groups(ctx, type):
+    """
+    List groups
+    """
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    logger: Logger = ctx.obj.get('logger')
+    response = api2.get_groups(type=type)
+    serialized_response = json.dumps(response, indent=2)
+    logger.info(f"{serialized_response}")
 
 
 @group.command('delete')
 @click.option('--name', "-n", required=True, type=str, help='Name of the group')
+@click.option('--user', 'type', flag_value='user_group')
+@click.option('--system', 'type', flag_value='system_group')
 @click.pass_context
-def delete_group(ctx, name):
+def delete_group(ctx, name, type):
     """
     Delete a group
     """
-    # FIXME: Ideally, this would output JSON info of the deleted group (similar to delete-user), but at the moment it's
-    #  unclear exactly how to do that, given the API wrappers that we have.
+    logger = ctx.obj.get('logger')
+    if type is None:
+        logger.error('groups must have a type (either "user" or "system")')
+        sys.exit(1)
+    api2 = JumpcloudApiV2(ctx.obj.get('key'))
+    group = api2.get_group(group_name=name, group_type=type)
+    if group is None:
+        logger.error(f"no group found of type '{type}', {name}")
+        sys.exit(1)
+    api2.delete_group(group['id'], type)
+    logger.info(f"successfully deleted group {name}")
+
+
+@group.command('add-user')
+@click.option('--name', "-n", required=True, type=str, help='name of the group')
+@click.option('--username', '-u', required=True, type=str, help='username of user to be added')
+@click.pass_context
+def add_user(ctx, name, username):
+    """
+    Add user to group (group type is assumed to be 'user_group')
```

Comment:
how about `Add a user to a Jumpcloud 'user' group`?

Reply:
fixed